File: create_test_image.py
from PIL import Image, ImageDraw

# Create image
img = Image.new('RGB', (600, 450), 'white')
draw = ImageDraw.Draw(img)

# Yellow box - AUTO
draw.rectangle([(40, 80), (160, 140)], fill='#FFFF99', outline='#FF6600', width=3)
draw.text((50, 95), 'AUTO', fill='black')

# Red box - REPAIR  
draw.rectangle([(190, 55), (370, 165)], fill='#FF9999', outline='#AA0000', width=3)
draw.text((210, 100), 'REPAIR', fill='black')

# Purple box - INVOICE
draw.rectangle([(385, 55), (605, 165)], fill='#BB99CC', outline='#6600AA', width=3)
draw.text((405, 100), 'INVOICE', fill='black')

# Invoice No
draw.rectangle([(170, 195), (350, 245)], fill='#FF9999', outline='#AA0000', width=2)
draw.text((180, 210), 'Invoice No.', fill='black')
draw.rectangle([(365, 195), (515, 245)], fill='#5577FF', outline='#0000AA', width=2)
draw.text((375, 210), '10000', fill='black')

# Invoice Date
draw.rectangle([(170, 270), (350, 320)], fill='#FF99CC', outline='#AA0055', width=2)
draw.text((180, 285), 'Invoice Date:', fill='black')
draw.text((375, 285), '01/12/2023', fill='black')

# Due Date
draw.rectangle([(170, 345), (350, 395)], fill='#99FF99', outline='#00AA00', width=2)
draw.text((180, 360), 'Due Date:', fill='black')
draw.text((375, 360), '02/12/2023', fill='black')

img.save('test_invoice.png')
print('Test image created: test_invoice.png')

File: server/create_test_image.js
// Simple test image creator that doesn't require external dependencies
// This creates a basic PNG with text using canvas-like drawing

const fs = require('fs');
const path = require('path');

// Simple PNG creator with text - we'll use the server's sharp module
const rootDir = require('path').dirname(__dirname);
const processImageWithServer = async () => {
  try {
    // Create an SVG string with our test content
    const svg = `<svg width="600" height="400" xmlns="http://www.w3.org/2000/svg">
      <rect width="600" height="400" fill="white"/>
      <rect x="40" y="80" width="120" height="60" fill="#FFFF99" stroke="#FF6600" stroke-width="3"/>
      <text x="50" y="115" font-size="24" font-family="Arial" fill="black" font-weight="bold">AUTO</text>
      <rect x="190" y="55" width="180" height="110" fill="#FF9999" stroke="#AA0000" stroke-width="3"/>
      <text x="210" y="115" font-size="24" font-family="Arial" fill="black" font-weight="bold">REPAIR</text>
      <rect x="385" y="55" width="220" height="110" fill="#BB99CC" stroke="#6600AA" stroke-width="3"/>
      <text x="405" y="115" font-size="24" font-family="Arial" fill="black" font-weight="bold">INVOICE</text>
      <rect x="40" y="200" width="150" height="50" fill="#FF9999" stroke="#AA0000" stroke-width="2"/>
      <text x="50" y="230" font-size="16" font-family="Arial" fill="black">Invoice No.</text>
      <rect x="200" y="200" width="150" height="50" fill="#5577FF" stroke="#0000AA" stroke-width="2"/>
      <text x="220" y="230" font-size="18" font-family="Arial" fill="black" font-weight="bold">10000</text>
      <rect x="40" y="260" width="150" height="50" fill="#FF99CC" stroke="#AA0055" stroke-width="2"/>
      <text x="50" y="290" font-size="16" font-family="Arial" fill="black">Amount:</text>
      <rect x="200" y="260" width="150" height="50" fill="#99FF99" stroke="#00AA00" stroke-width="2"/>
      <text x="220" y="290" font-size="18" font-family="Arial" fill="black" font-weight="bold">$1500.00</text>
    </svg>`;

    // Save SVG first
    const svgPath = path.join(__dirname, 'test_invoice.svg');
    fs.writeFileSync(svgPath, svg);
    console.log(`Created SVG: ${svgPath}`);
    
    // Now use server's sharp to convert
    const projectRoot = path.join(__dirname, '..');
    const serverModules = path.join(projectRoot, 'server', 'node_modules');
    const sharp = require(path.join(serverModules, 'sharp'));
    
    const outputPath = path.join(__dirname, 'test_invoice.png');
    await sharp(Buffer.from(svg))
      .png()
      .toFile(outputPath);
    
    console.log(`✓ Test image created: ${outputPath}`);
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
};

processImageWithServer();
